rfcs/0012-mtls.md

```diff
@@ -0,0 +1,106 @@
+
+## Enabling the Istio sidecar proxy in that component
+
+This can be done by adding the `sidecar.istio.io/inject=true` label to the component deployment template.
+The pod spec is updated dynamically by Istio to inject the `istio-proxy` sidecar into the pod.
+
+## Enabling mtls in the Wasm plugin when communicating with that component
+
+The corresponding EnvoyFilter for the component requires the following `transport_socket` configuration to be added:
+
+```yaml
+        transport_socket:
+          name: envoy.transport_sockets.tls
+          typed_config:
+            '@type': type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
+            common_tls_context:
+              tls_certificate_sds_secret_configs:
+              - name: default
+                sds_config:
+                  api_config_source:
+                    api_type: GRPC
+                    grpc_services:
+                    - envoy_grpc:
+                        cluster_name: sds-grpc
+              validation_context_sds_secret_config:
+                name: ROOTCA
+                sds_config:
+                  api_config_source:
+                    api_type: GRPC
+                    grpc_services:
+                    - envoy_grpc:
+                        cluster_name: sds-grpc
+```
+
+This configuration tells the gateway pod (envoy proxy) to use tls and validate certificates when talking with the upstream service (e.g. limitador or authorino). It also configures the location to get certificate information from i.e. the sds service/cluster in Istio.
+
+## Configuring the Istio PeerAuthentication resource to use STRICT mtls mode
+
+The appropriate operator should create a [PeerAuthentication](https://istio.io/latest/docs/reference/config/security/peer_authentication/) resource for either component to enable STRICT mtls mode.
+The resource can specify a pod label match so that it only takes effect for the authorino or limitador pod.
+For example:
+
+```yaml
+apiVersion: security.istio.io/v1
+kind: PeerAuthentication
+metadata:
+  name: limitador-mtls
+  namespace: kuadrant-system
+spec:
+  selector:
+    matchLabels:
+      app: limitador
+  mtls:
+    mode: STRICT
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Additional paths through the operator that result in a different configuration set of components to support.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Manual configuraton is an option (investigated in https://github.com/Kuadrant/kuadrant-operator/issues/1049 and documented in https://github.com/Kuadrant/kuadrant-operator/pull/1054)
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- What components will create and manage the resources and configuration detailed above?
+- What configuration options makes sense to bubble up via the Limitador and Authorino components, if any?
```

Comment:
Discussion around this happening -> https://groups.google.com/a/redhat.com/g/kuadrant-dev/c/rbsY4kDaSnE
And at the community call -> https://docs.google.com/document/d/1Dkcqu1MgCJlvJHF_iCYibPNsnSHu1aIeG5-WoqlKtGU/edit?tab=t.0#heading=h.5a78l4pgyj90
Will relay the outcome here via update with an answer to this question.